packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_list.dart

```diff
@@ -39,19 +41,28 @@ class _ListControlPane extends StatelessWidget {
     return ValueListenableBuilder<bool>(
       valueListenable: controller.isTakingSnapshot,
       builder: (_, isProcessing, __) {
-        final takeSnapshotEnabled = !isProcessing;
         final clearAllEnabled = !isProcessing && controller.hasSnapshots;
         return Row(
           children: [
             ToolbarAction(
               icon: Icons.fiber_manual_record,
               tooltip: 'Take heap snapshot for the selected isolate',
-              onPressed: takeSnapshotEnabled ? controller.takeSnapshot : null,
+              onPressed: controller.takeSnapshotHandler(
+                analytics_constants.MemoryEvent.diffTakeSnapshotControlPane,
+              ),
             ),
             ToolbarAction(
               icon: Icons.block,
               tooltip: 'Clear all snapshots',
-              onPressed: clearAllEnabled ? controller.clearSnapshots : null,
+              onPressed: clearAllEnabled
+                  ? () async {
+                      ga.select(
+                        analytics_constants.memory,
+                        analytics_constants.MemoryEvent.diffClearSnapshots,
+                      );
+                      await controller.clearSnapshots();
```

Comment:
should this be awaited? I don't think we want to block the UI on this finishing.